lib/elixir_search_extractor/file_upload.ex

```diff
@@ -0,0 +1,90 @@
+defmodule ElixirSearchExtractor.FileUpload do
+  import Ecto.Query, warn: false
+
+  alias ElixirSearchExtractor.FileUpload.KeywordFile
+  alias ElixirSearchExtractor.Repo
+
+  def list_keyword_files do
+    Repo.all(KeywordFile)
+  end
+
+  def create_keyword_file(attributes, user_id) do
+    csv_file = attributes["csv_file"]
+
+    case validate_file(csv_file) do
+      {:ok} ->
+        attributes = Map.put(attributes, "csv_file", upload_file_path(user_id, csv_file))
+        attributes = Map.put(attributes, "user_id", user_id)
+
+        %KeywordFile{}
+        |> KeywordFile.changeset(attributes)
+        |> Repo.insert()
+        |> upload_file(user_id, csv_file)
+
+      {:error, reason} ->
+        {:error, reason}
+    end
+  end
+
+  def change_keyword_file(%KeywordFile{} = keyword_file, attrs \\ %{}) do
+    KeywordFile.changeset(keyword_file, attrs)
+  end
+
+  defp validate_file(csv) do
+    with {:ok} <- validate_file_presence(csv),
+         {:ok} <- validate_file_size(csv),
+         {:ok} <- validate_extension(csv) do
+      {:ok}
+    else
+      err -> err
+    end
```

Comment:
I think we are good to just return `:ok` in this case and we can also remove else lines 👍 

```suggestion
      :ok
    end
```